lib/irb/ruby-lex.rb

```diff
@@ -65,68 +65,43 @@ def configure_io(io)
           else
             false
           end
-        else
+        elsif single_line_command?(code)
           # Accept any single-line input for symbol aliases or commands that transform args
-          next true if single_line_command?(code)
-
-          ltype, indent, continue, code_block_open = check_code_state(code)
-          if ltype or indent > 0 or continue or code_block_open
-            false
-          else
-            true
-          end
+          true
+        else
+          check_target_code, tokens, opens = check_code_state(code)
+          continue = process_continue(tokens)
+          opens.empty? && !continue && !check_code_block(check_target_code, tokens)
```

Comment:
I think these 3 lines could be extracted to a `code_terminated?` method? It can be use in `readmultiline` too and the intention would be clearer.

Reply:
In readmultiline, opens and tokens are also used after checking for code_termination, so I've done some other refactor that check_code_state also returns termination state